### PR TITLE
adds binding between users and projects they belong to

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: introduces a new resource UserProjectBinding that will store info about a user and a project he belongs to along with the group name within that project.
This helper resource will help us enforce "only projects owners can add/remove and delete members of the given project" semantics

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
#1966

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: api/pkg/crd/kubermatic/v1/user_project_binding.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//+genclient
+//+genclient:nonNamespaced
+
+// UserProjectBinding specifies a binding between a user and a project
+// This resource is used by the user management to manipulate members of the given project
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type UserProjectBinding struct {
```
Comment (p0lyn0mial) [inline api/pkg/crd/kubermatic/v1/user_project_binding.go:13]: This file is relevant the rest was generated.

Comment (p0lyn0mial) [inline api/pkg/crd/kubermatic/v1/user_project_binding.go:13]: I'll also add a proper yaml file under `config/kubermatic/templates` in a separate pull

Comment (p0lyn0mial) [reply api/pkg/crd/kubermatic/v1/user_project_binding.go:13]: @mrIncompetent is they anything else I need to add except the yaml file to make it to work ?

Comment (glower) [reply api/pkg/crd/kubermatic/v1/user_project_binding.go:13]: LGTM
but I will wait for a feedback from @mrIncompetent

**Review by mrIncompetent — APPROVED** [review]

File: api/pkg/crd/kubermatic/v1/user_project_binding.go (hunk 1 above)
Comment (mrIncompetent) [reply api/pkg/crd/kubermatic/v1/user_project_binding.go:13]: Adding the CRD to `config/kubermatic/templates` should be enough

**Comment by p0lyn0mial** [comment]
@glower @mrIncompetent added crd registration in this pull, PTAL
https://github.com/kubermatic/kubermatic/pull/1967

**Review by mrIncompetent — APPROVED** [review]

**Review by glower — APPROVED** [review]

**Pull request merged**